cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/MultipleOutputs.java

```diff
@@ -242,6 +245,47 @@ private static Job getNamedJob(JobContext context, String namedOutput) throws IO
     return job;
   }
 
+  /**
+   * Wraps RecordWriter to increment output counters.
+   */
+  @SuppressWarnings("unchecked")
+  private static class RecordWriterWithCounter extends RecordWriter {
+    private final RecordWriter writer;
+    private final String groupName;
+    private final String counterName;
+    private final TaskInputOutputContext context;
+
+    public RecordWriterWithCounter(RecordWriter writer, TaskInputOutputContext context) {
+      this.writer = writer;
+      this.context = context;
+      this.groupName = TaskCounter.class.getName();
+      this.counterName = getCounterName(context);
+    }
+
+    @SuppressWarnings({"unchecked"})
```

Comment:
Unsure actually. Maybe this is syntax for suppressing a set of warnings. Will remove it.